File: packages/core/dto/storageclass.go
package dto

type StorageClassRef struct {
	Name string `json:"name"`
}

type StorageClass struct {
	Name              string
	Provisioner       string
	ReclaimPolicy     string
	Default           bool
	Age               string
	CreatedAt         string
	Labels            map[string]string
	Annotations       map[string]string
	ManagedFields     []ManagedField
	VolumeBindingMode string
	MountOptions      []string
	Parameters        map[string]string
}

File: packages/core/frontend/src/types/resources/storageclass.ts
import type { ManagedField } from "./shared";

export interface StorageClass {
  Name: string;
  Provisioner: string;
  ReclaimPolicy: string;
  Default: boolean;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  VolumeBindingMode: string;
  MountOptions: string[];
  Parameters: Record<string, string>;
}
